test/AccessLock.t.sol

```diff
@@ -0,0 +1,726 @@
+            IPoolManager.ModifyPositionParams(0, 60, 1 * 10 ** 18),
+            abi.encode(amount, AccessLockHook.LockAction.Mint)
+        );
+
+        uint256 balanceOfAfter0 = MockERC20(Currency.unwrap(currency0)).balanceOf(address(this));
+        uint256 balanceOfAfter1 = MockERC20(Currency.unwrap(currency1)).balanceOf(address(this));
+
+        assertEq(balanceOfBefore0 - balanceOfAfter0, uint256(uint128(delta.amount0())));
+        // The balance of our contract should be from the modifyPositionRouter (delta) AND the hook (amount).
+        assertEq(balanceOfBefore1 - balanceOfAfter1, uint256(amount + uint256(uint128(delta.amount1()))));
+
+        assertEq(manager.balanceOf(address(accessLockHook), currency1), amount);
+    }
+
+    function test_beforeModifyPosition_take_succeedsWithAccessLock(uint128 amount) public {
+        // Add liquidity so there is something to take.
+        modifyPositionRouter.modifyPosition(
+            key,
+            IPoolManager.ModifyPositionParams({tickLower: -120, tickUpper: 120, liquidityDelta: 100 * 10e18}),
+            ZERO_BYTES
+        );
+        // Can't take more than the manager has.
+        vm.assume(amount < key.currency1.balanceOf(address(manager)));
+
+        uint256 balanceOfBefore1 = MockERC20(Currency.unwrap(currency1)).balanceOf(address(this));
+        uint256 balanceOfBefore0 = MockERC20(Currency.unwrap(currency0)).balanceOf(address(this));
+
+        // Hook only takes currency 1 rn.
+        BalanceDelta delta = modifyPositionRouter.modifyPosition(
+            key,
+            IPoolManager.ModifyPositionParams(-60, 60, 1 * 10 ** 18),
+            abi.encode(amount, AccessLockHook.LockAction.Take)
+        );
+        uint256 balanceOfAfter0 = MockERC20(Currency.unwrap(currency0)).balanceOf(address(this));
+        uint256 balanceOfAfter1 = MockERC20(Currency.unwrap(currency1)).balanceOf(address(this));
+
+        assertEq(balanceOfBefore0 - balanceOfAfter0, uint256(uint128(delta.amount0())));
+        // The balance of our contract should be from the modifyPositionRouter (delta) AND the hook (amount).
+        assertEq(balanceOfBefore1 - balanceOfAfter1, uint256(amount + uint256(uint128(delta.amount1()))));
+        assertEq(MockERC20(Currency.unwrap(currency1)).balanceOf(address(accessLockHook)), amount);
+    }
+
+    function test_beforeModifyPosition_swap_succeedsWithAccessLock(uint128 amount) public {
+        vm.assume(amount != 0 && amount > 10); // precision
+
+        // Add liquidity so there is something to swap over.
+        modifyPositionRouter.modifyPosition(
+            key,
+            IPoolManager.ModifyPositionParams({tickLower: -120, tickUpper: 120, liquidityDelta: 100 * 10e18}),
+            ZERO_BYTES
+        );
+
+        uint256 balanceOfBefore1 = MockERC20(Currency.unwrap(currency1)).balanceOf(address(this));
+        uint256 balanceOfBefore0 = MockERC20(Currency.unwrap(currency0)).balanceOf(address(this));
+
+        // Just no-op.
```

Comment:
nit: comment could be more descriptive since it's hard to know what logic is run on LockAction.Swap, it was a bit hard to understand what this test was doing. Like `// No-op the modifyPosition call and swap before hand, applying the deltas to the locker`